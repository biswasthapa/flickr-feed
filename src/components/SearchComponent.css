.search-container {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 100%;
  width: 100%;
  margin: 0 auto;
  min-height: 100%;
  min-height: 100vh;
}

.search-container h1 {
  color: #0063dc;
  font-size: 24px;
  margin: 0 0 10px;
}

.search-container h1 b {
  color: #fe0083;
}

.search-container .search-text {
  width: 100%;
  margin: 0 0 10px;
}

.search-container .btn {
  margin-top: 10px;
}

.search-container.search-top .btn {
  margin: 10px 0px 20px 10px;
}

.search-container.search-top {
  height: auto;
  width: 100%;
  flex-direction: row;
  min-height: 0;
}

.search-container.search-top h1 {
  font-size: 10px;
}

.search-block.search-top .input {
  border-radius: 0px;
  padding: 11px 30px;
}

.form-container {
  width: 100%;
  height: 100%;
}
