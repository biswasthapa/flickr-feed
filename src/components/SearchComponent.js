import React from "react";
import cx from "classnames";

import "./SearchComponent.css";

export const SearchComponent = ({ displayTop, onChange, onClick }) => {
  return (
    <div className={cx("search-container", { "search-top": displayTop })}>
      <h1>
        Search <b>Flickr!</b>
      </h1>
      <input
        type="text"
        className="form-control search-text"
        onChange={onChange}
        placeholder="Search using image tags..."
      />
      <span className="input-group-btn">
        <button
          className="btn btn-primary search-btn"
          type="submit"
          onClick={onClick}
        >
          Go!
        </button>
      </span>
    </div>
  );
};

export default SearchComponent;
