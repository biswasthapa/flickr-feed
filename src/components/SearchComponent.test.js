import React from "react";
import { SearchComponent } from "./SearchComponent";
import { shallow } from "enzyme";

describe("Search Component", () => {
  const defaultProps = {
    displayTop: false,
    onChange: jest.fn(),
    onClick: jest.fn()
  };

  it("should render", () => {
    const wrapper = shallow(<SearchComponent {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render search component at middle of page if displayTop is false", () => {
    const wrapper = shallow(<SearchComponent {...defaultProps} />);
    expect(wrapper.find("div.search-container")).toExist();
    expect(wrapper.find("div.search-container.search-top")).not.toExist();
  });

  it("should render search component at top of page if displayTop is true", () => {
    const props = { ...defaultProps, displayTop: true };
    const wrapper = shallow(<SearchComponent {...props} />);
    expect(wrapper.find("div.search-container.search-top")).toExist();
  });

  it("calls onChange function when someone starts typing in search input", () => {
    const wrapper = shallow(<SearchComponent {...defaultProps} />);
    wrapper
      .find("input.form-control.search-text")
      .simulate("change", { target: { value: "sydney" } });
    expect(defaultProps.onChange).toBeCalledTimes(1);
  });

  it("calls onClick function when someone submits the search", () => {
    const wrapper = shallow(<SearchComponent {...defaultProps} />);
    wrapper
      .find("input.form-control.search-text")
      .simulate("change", { target: { value: "sydney" } });
    wrapper.find("button.btn.btn-primary.search-btn").simulate("click");
    expect(defaultProps.onClick).toBeCalledTimes(1);
  });
});
