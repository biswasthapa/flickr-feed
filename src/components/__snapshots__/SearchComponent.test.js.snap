// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Component should render 1`] = `
<div
  className="search-container"
>
  <h1>
    Search 
    <b>
      Flickr!
    </b>
  </h1>
  <input
    className="form-control search-text"
    onChange={[MockFunction]}
    placeholder="Search using image tags..."
    type="text"
  />
  <span
    className="input-group-btn"
  >
    <button
      className="btn btn-primary search-btn"
      onClick={[MockFunction]}
      type="submit"
    >
      Go!
    </button>
  </span>
</div>
`;
